### PR TITLE
fix: multiple definitions for scap_engine_handle

### PR DESCRIPTION
**Pull request by federico-sysdig**
**What type of PR is this?**
/kind cleanup

**Any specific area of the project related to this PR?**
/area build

**Does this PR require a change in the driver versions?**
No

**What this PR does / why we need it**:
Class `scap_engine_handle` is defined once in its header file, but has a single pointer data member `SCAP_HANDLE_T* m_handle` that contains a macro that changes depending on what source file is including it. This is by design, the idea being that it is a sort of hierarchy of classes that share the same functions. However this is not without problems as if one tries to enable link-time optimization (LTO) the linker presents an error that the one-definition rule (ODR) has been broken.
This PR solves the issue by using a generic `void*` for the data member and using a cast to the appropriate type each time is is used in a function. This is possible as manipulating functions are specialized for each type, so knowledge of the type is available.
The technique sacrifices somewhat the "beauty" of the previous approach: each cast is syntactically "ugly", but saves the ODR and overcomes linker errors when LTO is enabled.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by federico-sysdig** [comment]
> We likely need a rebase for the CI to pass.

Thanks @incertum, I've rebased. Though I think there's some kind of issue that causes build problems in some scenarios. I will have to make time and sort them out.

**Comment by federico-sysdig** [comment]
I've recently rebased and now the required CI stages are passing. There are still some broken stages, but I can't see how they could be related to this PR.
The round of approvals need to restart after the rebase. Could anyone please review?

**Review by gnosek — APPROVED** [review]

**Comment by federico-sysdig** [comment]
Not sure whom to go to for soliciting the needed second review. Can anyone help, please?
cc @FedeDP @LucaGuerra @incertum

**Review by LucaGuerra — APPROVED** [review]

**Comment by federico-sysdig** [comment]
@hbrueckner
IIUC the system has elected you as one of the reviewer and your contribution is essential for moving the PR forward.

**Comment by incertum** [comment]
Not sure if it will merge, @LucaGuerra mind checking as I am offline now...

**Comment by incertum** [comment]
Kind bump, why is this stuck and not merging? @falcosecurity/libs-maintainers thanks!

**Comment by gnosek** [comment]
/approve

pretty please?

**Pull request merged**
